_posts/2020-08-07-cross-origin-cookies.md

```diff
@@ -0,0 +1,49 @@
+---
+title: SameSite Cookie Policies and DHIS2 Applications
+layout: post
+categories: [blog]
+tags: [app platform, developer tools, webapp, troubleshooting, authentication]
+authors: [austin]
+---
+
+As of mid-July 2020, the Chrome (and Chromium) stable release channel [has started to disable cross-site cookies by default](https://www.chromium.org/updates/same-site). Mozilla Firefox has pushed this change to their [beta channel and will likely release it to the stable channel soon](https://hacks.mozilla.org/2020/08/changes-to-samesite-cookie-behavior/). This change affects any DHIS2 application running **on a different domain than the DHIS2 server instance**, including applications running on localhost in development. It does not affect cross-site API requests which use Basic or OAuth authentication headers, as those do not rely on cookies for authentication.
+
+## What is the SameSite Cookie attribute?
+
+Basically, browsers are starting to block, by default, most cookies (which typically contain sensitive information such as login session tokens) to a domain which does not match the current URL of the site visited in the browser. The SERVER (not the browser) must set a specific attribute on the cookie (called `SameSite` to have the value `None-requires-secure`) if the cookie is safe to include in cross-domain requests (only over HTTPS connections!). DHIS2 does not set this attribute on its authentication cookies. As of mid-July, popular browsers like Chrome and Firefox are beginning to enforce these rules.
```

Comment:
I think the only values that the `SameSite` can take are 
1. `Lax` 
2. `Strict`
3. `None` 

We suggesting here the reader to set the value to `None-requires-secure`. What am I missing?

Reply:
Secure is a separate attribute on the cookie.  Chromium uses the `None-requires-secure` terminology and otherplaces it's written `SameSite=None; Secure`. I'll add a [link](https://web.dev/samesite-cookies-explained/#samesitenone-must-be-secure)